src/input_xml.F90

```diff
@@ -5628,17 +4632,206 @@ subroutine generate_rpn(cell_id, tokens, output)
   end subroutine generate_rpn
 
 !===============================================================================
-! NORMALIZE_AO normalizes the atom or weight percentages for each material
+! ASSIGN_NUCLIDE_DENSITIES Read in the nuclide densities from materials.xml
+! and assign to the corresponding nuclides.
+!===============================================================================
+
+  subroutine assign_nuclide_densities()
```

Comment:
I'm confused as to why nuclide densities are being read separately from the rest of the materials.xml file. Can you explain?